### PR TITLE
Clean up static analysis config

### PR DESCRIPTION
**Pull request by RDIL**
# PR Summary

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: .vsts-ci/misc-analysis.yml [hunk 1]
```diff
@@ -50,17 +51,14 @@ phases:
...
   - powershell: |
       Import-module ./build.psm1
       $path = Join-Path -Path $pwd -ChildPath './commonTestResults.xml'
       $results = invoke-pester -Script ./test/common -OutputFile $path -OutputFormat NUnitXml -PassThru
       Write-Host "##vso[results.publish type=NUnit;mergeResults=true;runTitle=Common Tests;publishRunAttachments=true;resultFiles=$path;]"
-      if($results.TotalCount -eq 0 -or $results.FailedCount -gt 0)
-      {
-        throw "markdown link tests failed"
-      }
+      if($results.TotalCount -eq 0 -or $results.FailedCount -gt 0) { throw "Common tests failed" }
```
Comment (TravisEz13) [inline .vsts-ci/misc-analysis.yml]: please change this back.
The standard is to have the code block for conditionals on a separate line.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: .vsts-ci/misc-analysis.yml (hunk 1 above)
Comment (RDIL) [reply .vsts-ci/misc-analysis.yml]: Fixed in 20391a3

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Windows build is set not to start when only static analysis is changed.

**Comment by RDIL** [comment]
Oh that’s why

**Comment by TravisEz13** [comment]
To me it is a bug that they don't report success...  The fact that I told it not to run in this case, means everything is good.

File: .vsts-ci/misc-analysis.yml [hunk 2]
```diff
@@ -60,7 +61,7 @@ phases:
       Write-Host "##vso[results.publish type=NUnit;mergeResults=true;runTitle=Common Tests;publishRunAttachments=true;resultFiles=$path;]"
       if($results.TotalCount -eq 0 -or $results.FailedCount -gt 0)
       {
-        throw "markdown link tests failed"
+        throw "Common tests failed"
```
Comment (daxian-dbw) [inline .vsts-ci/misc-analysis.yml]: "Common tests failed" is a little confusing without knowing that means the tests under `./test/common` folder. Maybe it's better say `Tests in './test/common' failed`.

Comment (RDIL) [reply .vsts-ci/misc-analysis.yml]: Changed to `markdown`

**Review by TravisEz13 — APPROVED** [review]

**Comment by RDIL** [comment]
@TravisEz13 anything blocking?

**Pull request merged**